hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -118,6 +118,12 @@ private[hudi] object HoodieSparkSqlWriter {
         fs.delete(basePath, true)
         exists = false
       }
+      if (exists && mode == SaveMode.Append) {
```

Comment:
I think this check is applicable to both the if and else clause. Can you verify that and move it to may be line [85](https://github.com/apache/incubator-hudi/pull/1580/files#diff-5317f4121df875e406876f9f0f012facR85) (before the if clause) ?

Reply:
Thanks for the comment. I have updated the code.

Reply:
@bhasudha please review! 

Thanks!